README.md

```diff
@@ -34,14 +34,236 @@ VSCode allows for development within a docker container. This project has an alr
+    - clone the repository using SSH:
+        `git clone git@github.com:team/repository/.git`
+
+### Local Development
+
+1. Install project dependencies
+    `make install`
+2. Run the development environment:
+    if using Visual Studio Code: click the "Reopen in Container" option in the bottom left conner. This will set up the development environment using Docker.
+
+### Running the app in container:
+
+To run the app in the container:
+1. For the API app:
+    `make run`
+2. Using Docker:
+    `docker -compose up --build`
+3. For the CLI commands:
+    `poetry run darkmoon --help`
+
+### MongoDB Compass
+
+The database that is used for this project is MongoDB. The file metadata is uploaded to the MongoDB database after the program is run. MongoDB allows for easier lookup and sorting of file metadata.
+
+-[MongoDB](https://www.mongodb.com)
+
+To connect to the MongoDB database used by the API:
+1. Launch MongoDB Compass.
+2. Use the connection string specified in the `settings.py` file to connect to the database.
+
+# DarkMoon CLI
+
+The DarkMoon CLI subsection of the project scans operating systems, iterates through the files, and sends them through to MongoDB via FastAPI.
+Below is a detailed list of the functions found in this section, their arguments, results, and purpose.
+
+## Main Functions
+
+These are the most high-level functions that you will mostly be interacting with.
+
+### get_metadata
+
+Calls all of the minor metadata-collecting functions in one place and formats it appropriately.
+
+**Arguments:**
+
++ File Path (Path)
++ ISO Name (String)
++ Debug (Boolean) | OPTIONAL: Default is FALSE
+
+**Result:**
+
++ Dictionary (string:Any) with the file metadata formatted for FastAPI POST request
+
+**Example:**
+
+
+### iterate_files
+
+Iterates through the operating systems, obtains the file data, formats the data, and sends it to FastAPI.
+
+**Arguments:**
+
++ File Path (Path)
++ ISO Name (String)
++ Debug (Boolean) | OPTIONAL: Default is FALSE
+
+**Results**
+
++ Returns "None", but sends data over to FastAPI
+
+**Example:**
+
+
+## Important Functions
+
+While you won't really be interacting directly with these functions, they play an important role in the program.
+
+### call_api
+
+Sends a dictionary to FastAPI. THIS COMMAND CAN _NOT_ BE CALLED IN THE CONSOLE!
+
+**Arguments:**
+
++ Dictionary
+
+**Results:**
+
++ Returns nothing; sends the data over to FastAPI
+
+**Example:**
+
+
+### get_all_exe_metadata
+
+Obtains all the additional metadata needed for .exe and .dll files
+
+**Arguments:**
+
++ File Path (Path)
++ Debug (Boolean) | OPTIONAL: Default is FALSE
+
+**Results:**
+
++ A dictionary (string:Any) with all the metadata, including the unique types for .exe and .dll files, is returned
+
+**Example:**
+
+
+### get_file_type
+
+Gets the file type of a file
+
+**Arguments:**
+
++ File Path (Path)
+
+**Results:**
+
++ Returns the file type (the first word from the string returned by the function)
+
+**Example:**
```

Comment:
I'm going to say this readme should not just copy and paste any docstrings. A readme is to explain how to use it, if you're going to explain every command it has then give a short description and a use case.

Otherwise group them together using `poetry run darkmoon --help`